tests/Dialect/Field/quartic_ext_field_to_mod_arith.mlir

```diff
@@ -0,0 +1,107 @@
+// Copyright 2025 The ZKIR Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific Language governing permissions and
+// limitations under the License.
+// ==============================================================================
+
+// RUN: zkir-opt -field-to-mod-arith %s | FileCheck %s
+
+#xi = #field.pf.elem<2:i32> : !field.pf<7:i32>
+!PF = !field.pf<7:i32>
+!PFm = !field.pf<7:i32, true>
+#xi_mont = #field.pf.elem<1:i32> : !PFm
+!QF = !field.f4<!PF, #xi>
+!QFm = !field.f4<!PFm, #xi_mont>
+
+// CHECK-LABEL: @test_lower_add
+func.func @test_lower_add(%arg0: !QF, %arg1: !QF) -> !QF {
+    // CHECK-COUNT-2: field.ext_to_coeffs
+    // CHECK-COUNT-4: mod_arith.add
+    // CHECK: field.ext_from_coeffs
+    %0 = field.add %arg0, %arg1 : !QF
+    return %0 : !QF
+}
+
+// CHECK-LABEL: @test_lower_sub
+func.func @test_lower_sub(%arg0: !QF, %arg1: !QF) -> !QF {
+    // CHECK-COUNT-2: field.ext_to_coeffs
+    // CHECK-COUNT-4: mod_arith.sub
+    // CHECK: field.ext_from_coeffs
+    %0 = field.sub %arg0, %arg1 : !QF
+    return %0 : !QF
+}
+
+// CHECK-LABEL: @test_lower_mul
+func.func @test_lower_mul(%arg0: !QF, %arg1: !QF) -> !QF {
+    // Toom-Cook multiplication uses 6 base field multiplications
+    // CHECK: mod_arith.constant
+    // CHECK-COUNT-2: field.ext_to_coeffs
+    // CHECK: mod_arith.mul
```

Comment:
Could you count the mul more strictly, so that we can compare the performance if there's a better algorithm